Spectator will follow the target into other dimensions

EpyonProjects opened the pull request.

EpyonProjects committed:
Teleport spectator to dimensions with the target
- Additionally, add return to break out of watch self

EpyonProjects committed:
Fixed an unused warning

kruthers commented on src/main/kotlin/com/kruthers/gamemode4core/events/DimensionEvents.kt:
```diff
@@ -22,6 +23,17 @@ class DimensionEvents(val plugin: Gamemode4Core): Listener {
                     if (target == player.uniqueId) {
+                        watcher.spectatorTarget?.let {
+                            watcher.spectatorTarget = null
+                            Bukkit.getScheduler().runTaskLater(plugin, Runnable {
```
instead of delaying 1 tick after you can get `event.to` and tp to there immediately. if you adjust the listener and check its not cancled. Then you only need 1 delay to make them a specator them again.

kruthers commented on src/main/kotlin/com/kruthers/gamemode4core/modes/ModMode.kt:
```diff
@@ -52,6 +54,18 @@ class ModMode {
+            }
+
+            // Load their build mode effects
+            playerData.getList("storage.mod_mode.build_data.effects")?.forEachIndexed { _, it ->
```
this can just be `?.forEach { }` you wont need to define `it` either

EpyonProjects replied:
I'm regretting making 2 PRs at the same time haha.  Hopefully this merges properly with #3 being accepted first?  Otherwise if it's easier, I can close #3 and just add the change here

EpyonProjects committed:
Removed additional scheduled task
- removed check using map.values because it is also O(n), so it accomplishes the same by only doing the for each and just skipping with return@forEach

kruthers committed:
Merge branch 'main' into modmode-spectate-dimensional-travel
